### PR TITLE
doc: add no_contract flag to index

### PR DESCRIPTION
**Pull request by bennn**

### CODE REVIEW

File: typed-racket-doc/typed-racket/scribblings/reference/optimization.scrbl [hunk 1]
```diff
@@ -36,4 +36,5 @@ untyped code.
 By default, these contracts do not check that typed code obeys the types.
 If you want to generate contracts that check both sides equally (for analysis,
 for teaching, etc.) then set the environment variable
-@envvar{PLT_TR_NO_CONTRACT_OPTIMIZE} to any value and recompile.
+@index["PLT_TR_NO_CONTRACT_OPTIMIZE"]{@envvar{PLT_TR_NO_CONTRACT_OPTIMIZE}}
```
Comment (samth) [inline typed-racket-doc/typed-racket/scribblings/reference/optimization.scrbl:39]: Seems like this should be supported directly in `envvar`.

Comment (bennn) [reply typed-racket-doc/typed-racket/scribblings/reference/optimization.scrbl:39]: by default, or with an optional `#:index?` flag?

**Comment by samth** [comment]
Probably optional because we don't want to index every place PATH is referenced.

**Comment by samth** [comment]
I'm going to merge this, but I do think adding to `envvar` is a good idea.

**Comment by bennn** [comment]
oh, thanks for merging!

I'll try to update envvar this month.

**Pull request merged**